source/http/source.go

```diff
@@ -189,6 +192,24 @@ func (hs *httpSourceHandler) CacheKey(ctx context.Context, g session.Group, inde
 	req.Header.Add("User-Agent", version.UserAgent())
 	m := map[string]cacheRefMetadata{}
 
+	// Add an Authorization header if an HTTP auth header secret was defined
```

Comment:
The `urlHash` should either contain the name of the secret or at least boolean if authorization was added or not.

Reply:
Done. I also added it to `formatCacheKey`. 

Following that, I realized that in the case of a provided `Checksum`, no auth header would be added since the request is performed in `Snapshot`. I went ahead and added a commit that first refactors the `http.Request` creation into a common method.

Now to figure out if I can simply rebase the HTTP headers change on that commit... (I work mostly in Gerrit so I'm used to a stacked patchset workflow.)